cmake-d/UseDUnittest.cmake

```diff
@@ -7,31 +7,41 @@
 # See LICENSE for details.
 #
 
-macro(add_unittests _sourcefile)
-
-  set(_testname "${_sourcefile}")
-  set(main_unittest "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/unittest.d")
-  set(_source_file "${CMAKE_CURRENT_SOURCE_DIR}/${_sourcefile}")
-
-  #message("main_unittest ${main_unittest}")
-  #message("testname ${_testname}")
-  #message("_sourcefile ${_source_file}")
-
-  if(NOT EXISTS ${main_unittest})
-    file(WRITE ${main_unittest} "// Generated by UseDUnittest.cmake\nint main() { return 0; }")
-  endif()
-
-  # TODO
-  # better way to get the included directories
-  get_directory_property(include_dirs INCLUDE_DIRECTORIES)
-  set(INCLUDES )
-  foreach(include_dir IN LISTS include_dirs)
-    list(APPEND INCLUDES "${CMAKE_INCLUDE_FLAG_D}${include_dir}")
-  endforeach()
-
-  separate_arguments(CMAKE_D_FLAGS)
-
-  add_test(NAME ${_testname}
-  	COMMAND ${CMAKE_D_COMPILER} ${CMAKE_D_FLAGS} ${INCLUDES} ${main_unittest} -unittest -run ${_source_file})
-
+macro(target_add_unittests _target _sourceFileList)
```

Comment:
Maybe a better name `add_unittests_by_target` or `add_unittests_reuse_compile_flags`

Reply:
I chose targert_add_unittest to stay in line with calls like target_link_libraries or target_include_directories. It seems all target related stuff begins with target_

Reply:
Yes, but these commands change the properties of a target. But in your case **new** targets are created. It is more related to `add_test` or `gtest_add_test`.

Reply:
ok, so add_unittests_by_target then? or maybe add_unittests_for_target I think would better fit? Also I would like to reuse also the sources, so the call would look like add_unittests_for_target(target). Just didn't figure out yet how to get the sourcefiles of the target.